collections-csharp/ConcurrentQueueInCSharp/ConcurrentQueueInCSharp.Console/Producer.cs

```diff
@@ -0,0 +1,27 @@
+﻿using ConcurrentQueueInCSharp.Models;
+
+namespace ConcurrentQueueInCSharp
+{
+    public class Producer
+    {
+        private readonly OrderMessageBus _messageBus;
+        private readonly int _numberOfMessages;
+
+        public Producer(OrderMessageBus messageBus, int numberOfMessages)
+        {
+            _messageBus = messageBus;
+            _numberOfMessages = numberOfMessages;
+        }
+
+        public Task Produce()
+        {
+            return Task.Factory.StartNew(() =>
+            {
+                for (int i = 0; i < _numberOfMessages; i++)
+                {
+                    _messageBus.Add(new Order { Id = Guid.NewGuid().ToString() });
```

Comment:
If we modify order to have create the Id itself (which would often be a typical pattern), we can simplify this line.

Reply:
I personally wouldn't do this. Normally I would create some kind of factory class for `Order` that would take care of it since I could be using the `Order` class to retrieve existing orders from the database and I want to maintain the same order id.